kitsune/questions/feeds.py

```diff
@@ -13,7 +13,23 @@
 from kitsune.sumo.templatetags.jinja_helpers import urlparams
 
 
-class QuestionsFeed(Feed):
+class SyncFeed(Feed):
+    """
+    For some reason when running via the Meinheld WSGI worker for Gunicorn, the async
+    nature of the response cycle doesn't properly handle responses that have more than
+    one item in the `Response._container` list. This class squashes that list before
+    returning the HttpResponse object.
+
+    See https://github.com/mozilla/kitsune/issues/3017
+    """
+    def __call__(self, request, *args, **kwargs):
+        response = super(SyncFeed, self).__call__(request, *args, **kwargs)
+        # this squashes the response._container list to a single item
+        response.content = response.content
```

Comment:
I wouldn't have realized that this did anything at all without the comment above. Interesting.

Reply:
But _how_ is the above working?

Reply:
Yeah. It's weird. In the `HttpResponse` object `response.content` [calls a property method](https://github.com/django/django/blob/stable/1.8.x/django/http/response.py#L354) that returns a join of all the items in the `_container` list. [Assigning to that property](https://github.com/django/django/blob/stable/1.8.x/django/http/response.py#L367) puts the given string into a list of length 1.

Reply:
The code isn't clear, but it is the most succinct way of reducing that list to length 1 using the built-in methods.